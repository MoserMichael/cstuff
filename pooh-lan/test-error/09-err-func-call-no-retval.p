


sub no_return_value()
  println( ~msg 'function does not return a value' )
end


a = 2 * no_return_value()
