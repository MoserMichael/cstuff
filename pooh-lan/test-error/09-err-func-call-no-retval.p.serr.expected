
test-error/09-err-func-call-no-retval.p(9): Error: Function no_return_value does not return a value, but is used as any type
      9 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |a = 2 * no_return_value()
        |........^^^^^^^^^^^^^^^^^

test-error/09-err-func-call-no-retval.p(4): Error: Function no_return_value is defined here
      4 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |sub no_return_value()
        |^^^^^^^^^^^^^^^^^^^^^
Type checking failed
